db/indextest/util_testing.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2020-Present Couchbase, Inc.
+
+Use of this software is governed by the Business Source License included in
+the file licenses/BSL-Couchbase.txt.  As of the Change Date specified in that
+file, in accordance with the Business Source License, use of this software will
+be governed by the Apache License, Version 2.0, included in the file
+licenses/APL2.txt.
+*/
+
+package indextest
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/db"
+	"github.com/stretchr/testify/require"
+)
+
+func getDatabaseContextOptions(isServerless bool) db.DatabaseContextOptions {
+	defaultCacheOptions := db.DefaultCacheOptions()
+
+	return db.DatabaseContextOptions{
+		CacheOptions: &defaultCacheOptions,
+		Serverless:   isServerless,
+	}
+}
+
+func setupTestDBForBucketWithOptions(t testing.TB, dbcOptions db.DatabaseContextOptions) (*db.Database, context.Context) {
```

Comment:
For this in particular, I'd create a public version in db/util_testing.go for shared use by the db and db/indextest packages.